workflow/executor/common/wait/wait.go

```diff
@@ -0,0 +1,45 @@
+package wait
+
+import (
+	"fmt"
+
+	log "github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/argoproj/argo/workflow/executor/common"
+)
+
+func UntilTerminated(kubernetesInterface kubernetes.Interface, namespace, podName, containerID string) error {
+	log.Infof("Waiting for container %s to complete", containerID)
+	listOptions := metav1.ListOptions{FieldSelector: "metadata.name=" + podName}
+	for {
+		done, err := untilTerminatedAux(kubernetesInterface, namespace, containerID, listOptions)
+		if done {
+			return err
+		}
+	}
+}
+
+func untilTerminatedAux(kubernetesInterface kubernetes.Interface, namespace, containerID string, listOptions metav1.ListOptions) (bool, error) {
+	w, err := kubernetesInterface.CoreV1().Pods(namespace).Watch(listOptions)
```

Comment:
I looked at using `watch.Until` from the `go-client`, but it requires `resourceVersion` to work at all. There is `ListAndWatchUntil`, but this requires `list pods` but `workflow-role` does not have that permission.